org.eclipse.xtext/src/org/eclipse/xtext/EcoreUtil2.java

```diff
@@ -773,4 +775,58 @@ protected EObject computeNext() {
 			}
 		};
 	}
+	
+	public static URI getPathURI(final EObject object) {
```

Comment:
maybe call this `getIndexBasedPathURI` or `getNameAgnosticURI` since the segments of n URI always form a *path* and the term *path* does not express that the URI is agnostic of names.